File: java/eyes.playwright.java/src/main/java/com/applitools/eyes/playwright/universal/dto/AccessibilityRegionByElement.java
package com.applitools.eyes.playwright.universal.dto;

import com.applitools.eyes.fluent.GetRegion;
import com.applitools.eyes.universal.dto.TAccessibilityRegion;
import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class AccessibilityRegionByElement extends TAccessibilityRegion implements GetRegion {

    private Element region;

    public Element getRegion() {
        return region;
    }

    public void setRegion(Element region) {
        this.region = region;
    }

}

File: dotnet/Eyes.Playwright.DotNet/Fluent/AccessibilityRegionByElement.cs
using Applitools.Fluent;
using Applitools.Playwright.Universal.Dto;
using Applitools.Universal;
using Microsoft.Playwright;
using Refer = Applitools.Playwright.Universal.Refer;

namespace Applitools.Playwright.Fluent
{
    /// <summary>
    /// Used internally to represent a ref of an accessibility element
    /// </summary>
    public class AccessibilityRegionByElement : Element, IGetAccessibilityRegion, IGetAccessibilityRegionType, IPlaywrightReference<TAccessibilityRegion>
    {
        public AccessibilityRegionType AccessibilityRegionType { get; }

        public AccessibilityRegionByElement(IElementHandle element, AccessibilityRegionType type) 
            : base(element)
        {
            AccessibilityRegionType = type;
        }

        public new TAccessibilityRegion ToRegion()
        {
            return new ElementAccessibilityRegion
            {
                Type = AccessibilityRegionType,
                Region = new RegionElement
                {
                    //ElementId = base.ElementHandle
                }
            };
        }

        TAccessibilityRegion IPlaywrightReference<TAccessibilityRegion>.ToRegion(Reference root, Refer refer)
        {
            return new ElementAccessibilityRegion
            {
                Type = AccessibilityRegionType,
                Region = new RegionElement
                {
                    ElementId = refer.Ref(ElementHandle, root)
                }
            };
        }
    }
}